test/e2e_node/mirror_pod_grace_period_test.go

```diff
@@ -131,6 +131,26 @@ var _ = SIGDescribe("MirrorPodWithGracePeriod", func() {
 			framework.ExpectEqual(pod.Spec.Containers[0].Image, image)
 		})
 
+		ginkgo.It("should terminate a static pod when the container runtime is temporarily down [NodeConformance] [Serial]", func() {
+			ginkgo.By("delete the static pod")
+			err := deleteStaticPod(podPath, staticPodName, ns)
+			framework.ExpectNoError(err)
+
+			ginkgo.By("sleeping before stopping the container runtime")
+			time.Sleep(2 * time.Second)
+
+			ginkgo.By("stop the container runtime")
+			err = stopContainerRuntime()
+			framework.ExpectNoError(err, "expected no error stopping the container runtime")
```

Comment:
Reading through this test, where do we verify that the pod doesn't stop *while* the contanier runtime is down?

Naively reading this, I'm thinking I expected:

1. Create static pod
2. Stop container runtime
3. Delete static pod
4. Test for X seconds that the pod doesn't go away
5. Start container runtime
6. Verify static pod goes away?

I'm ask this because I'm adding metrics and I'm looking for a hook place to put the metrics check, and didn't see what I expected, so want to understand the test first.

Reply:
That makes sense of something we should check as well. I didn't add it originally as the test served more as regression test (i.e without https://github.com/kubernetes/kubernetes/pull/113145 the pod would never be able to deleted thus failing the test).

 Pushed latest revision that adds another block to verify that the mirror pod is running while the container runtime is down.